spec.html

```diff
@@ -33835,29 +33810,34 @@ <h1>%TypedArray%.prototype.sort ( _comparefn_ )</h1>
           1. Let _obj_ be the *this* value.
           1. Let _buffer_ be ? ValidateTypedArray(_obj_).
           1. Let _len_ be _obj_.[[ArrayLength]].
+          1. Let _objIsSparse_ be *false*.
         </emu-alg>
         <p>The implementation-defined sort order condition for exotic objects is not applied by %TypedArray%`.prototype.sort`.</p>
-        <p>The following version of SortCompare is used by %TypedArray%`.prototype.sort`. It performs a numeric comparison rather than the string comparison used in <emu-xref href="#sec-array.prototype.sort"></emu-xref>. SortCompare has access to the _comparefn_ and _buffer_ values of the current invocation of the `sort` method.</p>
-        <p>When the TypedArray SortCompare abstract operation is called with two arguments _x_ and _y_, the following steps are taken:</p>
-        <emu-alg>
-          1. Assert: Both Type(_x_) and Type(_y_) is Number.
-          1. If _comparefn_ is not *undefined*, then
-            1. Let _v_ be ? ToNumber(? Call(_comparefn_, *undefined*, &laquo; _x_, _y_ &raquo;)).
-            1. If IsDetachedBuffer(_buffer_) is *true*, throw a *TypeError* exception.
-            1. If _v_ is *NaN*, return *+0*.
-            1. Return _v_.
-          1. If _x_ and _y_ are both *NaN*, return *+0*.
-          1. If _x_ is *NaN*, return 1.
-          1. If _y_ is *NaN*, return -1.
-          1. If _x_ &lt; _y_, return -1.
-          1. If _x_ &gt; _y_, return 1.
-          1. If _x_ is *-0* and _y_ is *+0*, return -1.
-          1. If _x_ is *+0* and _y_ is *-0*, return 1.
-          1. Return *+0*.
-        </emu-alg>
-        <emu-note>
-          <p>Because *NaN* always compares greater than any other value, *NaN* property values always sort to the end of the result when _comparefn_ is not provided.</p>
-        </emu-note>
+        <p>%TypedArray%`.prototype.sort` calls TypedArraySortCompare rather than SortCompare.</p>
+
+        <emu-clause id="sec-typedarraysortcompare" aoid="TypedArraySortCompare">
+          <h1>Runtime Semantics: TypedArraySortCompare ( _x_, _y_ )</h1>
+          <p>The TypedArraySortCompare abstract operation is called with two arguments _x_ and _y_. It also has access to the _comparefn_ and _buffer_ values of the current invocation of the %TypedArray%`.prototype.sort` method. It performs a numeric comparison rather than the string comparison used in <emu-xref href="#sec-array.prototype.sort"></emu-xref>. The following steps are taken:</p>
+          <emu-alg>
+            1. Assert: Both Type(_x_) and Type(_y_) is Number.
```

Comment:
```suggestion
            1. Assert: both Type(_x_) and Type(_y_) are Number.
```

Reply:
(Note that the `<emu-alg>` text is the same as in the current spec -- github is only showing it as different because I had to change the indentation.)

re lowercase 'b': The spec is pretty consistent in capitalizing the first word after "Assert:".

re "is" to "are": That would be better, but I think I'd prefer `Assert: Type(_x_) is Number and Type(_y_) is Number`

Reply:
Sounds good on both points.